common/kvstore/src/main/java/org/apache/spark/util/kvstore/LevelDBIterator.java

```diff
@@ -280,4 +302,43 @@ static int compare(byte[] a, byte[] b) {
     return a.length - b.length;
   }
 
+  static class ResourceCleaner implements Runnable {
+
+    private final DBIterator dbIterator;
+
+    private final LevelDB levelDB;
+
+    private final AtomicBoolean started = new AtomicBoolean(true);
+
+    ResourceCleaner(DBIterator dbIterator, LevelDB levelDB) {
+      this.dbIterator = dbIterator;
+      this.levelDB = levelDB;
+    }
+
+    @Override
+    public void run() {
+      if (started.compareAndSet(true, false)) {
+        levelDB.notifyIteratorClosed(dbIterator);
+        synchronized (levelDB.getLevelDB()) {
+          DB _db = levelDB.getLevelDB().get();
+          if (_db != null) {
+            try {
+              dbIterator.close();
+            } catch (IOException e) {
+              throw new UncheckedIOException(e);
```

Comment:
nit: Personally: Perhaps printing a log is enough, as there should be no other places capturing this exception now.

Reply:
Good suggestion, I re-examined it，the JniDBIterator  can't throw an IOException, could we just ignore it?

<img width="632" alt="image" src="https://github.com/apache/spark/assets/49054376/4c5c9e6c-4c06-4175-9a1f-e12577a6f86a">